### DISCUSSION TITLE
How do I use EOF in vim.system?

### DISCUSSION THREAD

**Discussion by Demianeen** [neovim/neovim#30360] [Q&A] [answered]
I would like to pass a text from neovim to the command through <<EOF
```lua
vim.system({
          'silicon',
          '--to-clipboard',
          '-l',
          vim.bo.filetype,
          [[ <<EOF
]] .. text .. [[
EOF
]],
        }, { text = true }, on_exit)
```

But the command itself gives stderr `[error] No such file or directory (os error 2)`. I tried manually in terminal and everything works fine. So, I assume the problem is with table of arguments for system. 

Did anyone used vim.system with EOF?

**Comment by zeertzjq** [comment, marked as answer]
You don't need that. You can just pass the text directly, since there is no shell.
